### PR TITLE
feat: add the Trace Explorer page with Query Builder

### PR DESCRIPTION
**Pull request by dnazarenkoo**
Part of this epic - [2315](https://github.com/SigNoz/signoz/issues/2315)
Closes these tickets - [2332](https://github.com/SigNoz/signoz/issues/2332) & [2335](https://github.com/SigNoz/signoz/issues/2335)

### CODE REVIEW

File: frontend/src/pages/TracesExplorer/index.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import { CURRENT_TRACES_EXPLORER_TAB, initialTracesQuery } from './constants';
+import { Container } from './styles';
+import { getTabsItems } from './utils';
+
+function TracesExplorer(): JSX.Element {
+	const { currentQuery, redirectWithQueryBuilderData } = useQueryBuilder();
+
+	const urlQuery = useUrlQuery();
+	const history = useHistory();
+	const location = useLocation();
```
Comment (palashgdev) [inline frontend/src/pages/TracesExplorer/index.tsx:19]: from `react-router`

File: frontend/src/pages/TracesExplorer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	useEffect(() => {
+		const currentUrlTab = urlQuery.get(CURRENT_TRACES_EXPLORER_TAB);
+
+		if (currentUrlTab) {
+			setCurrentTab(currentUrlTab);
+		} else {
+			redirectWithCurrentTab(currentTab);
+		}
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (palashgdev) [inline frontend/src/pages/TracesExplorer/index.tsx:61]: can we remove this one and updated the dependency

File: frontend/src/types/api/trace/getQueryRange.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import { DataSource } from 'types/common/queryBuilder';
+
+import { BuilderQueryDataResourse } from '../queryBuilder/queryBuilderData';
+import { QueryData, QueryDataV3 } from '../widgets/getQuery';
+
+export type TracesExplorerRangeProps = {
+	start: number;
+	end: number;
+	step: number;
+	dataSource: DataSource;
```
Comment (palashgdev) [inline frontend/src/types/api/trace/getQueryRange.ts:12]: this is already created no need for this one

**Review by palashgdev — CHANGES_REQUESTED** [review]

**Review by palashgdev — DISMISSED** [review]

**Review by palashgdev — DISMISSED** [review]

**Review by palashgdev — APPROVED** [review]

**Comment by makeavish** [comment]
Found few issues @dnazarenkosignoz 
1. I see duplicate attribute keys and attribute values API calls when using the filter. (Not related to this PR)
2. Unnecessary aggregate key API calls are being made when clicking on the aggregate attribute field. (Not related to this PR)
3. Stuck on key suggestions API after pressing backspace: https://share.vidyard.com/watch/Sxw3HYQEH6kvXZAKHy1gVR? (Not related to this PR)
4. Unnecessary attribute key call when updating aggregate operator (Not related to this PR)

These issues can be addressed separately: https://github.com/SigNoz/signoz/issues/2932 & https://github.com/SigNoz/signoz/issues/2934

**Review by makeavish — APPROVED** [review]

**Pull request merged**